### PR TITLE
Update maven flexpack issues

### PR DESCRIPTION
**Pull request by agrasth**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] Appropriate label is added to auto generate release notes.
- [x] I used gofmt for formatting the code before submitting the pull request.
- [x] PR description is clear and concise, and it includes the proposed solution/fix.
-----
# Maven FlexPack: Fix Critical XML Parsing and Artifact Collection Issues

### 1. Added Proper XML Parsing
- Added structs: `PomProject`, `PomParent`, `DistributionManagement`, `Repository`
- Replaced `strings.Index()` XML parsing with `xml.Unmarshal()`
- Imports: Added `encoding/xml`

**Issue**: String-based parsing breaks with XML comments, whitespace, namespaces, and attribute reordering.

### 2. Fixed Deploy Goal Detection
- Updated `wasDeployCommand()` to match both `deploy` and `*:deploy` patterns

**Issue**: Maven plugin notation `maven-deploy-plugin:deploy` was not detected, causing artifact collection to be skipped.

### 3. Fixed Artifact Collection
- Replaced hardcoded `.jar` lookup with directory scan
- Now collects all artifacts: `.jar`, `.war`, `.ear`, `-sources.jar`, `-javadoc.jar`, `-tests.jar`
- Added `getPackagingType()` helper

**Issue**: Only main JAR was collected, missing sources, javadoc, tests, and other packaging types (war/ear).

### 4. Fixed Repository Detection
- `getMavenDeployRepository()` now uses XML parsing
- Handles both `<repository>` and `<snapshotRepository>`
- Supports two URL formats: standard and `/api/maven/` patterns

**Issue**: String parsing failed on real-world POM files, and snapshots weren't handled correctly.

### 5. Added Settings.xml Support for Deployment Repository
Problem: Users who define deployment repositories in settings.xml via altDeploymentRepository weren't supported - only pom.xml was checked.
**Fix:**
- Added SettingsXml struct parsing with active profile support
- Finds settings.xml via -s flag or ~/.m2/settings.xml
- Extracts altDeploymentRepository from active profiles (format: id::layout::url)
- Falls back to settings.xml only if pom.xml has no <distributionManagement>

## Why These Were Missed

These issues weren't caught because:
- Initial implementation was tested with minimal/simple POM files
- Tests didn't cover plugin notation, multi-artifact projects, or snapshot deployments
- String-based XML parsing worked for the happy path but failed on edge cases

### CODE REVIEW

File: artifactory/commands/flexpack/maven.go [hunk 1]
```diff
@@ -144,63 +180,159 @@ func setMavenBuildPropertiesOnArtifacts(workingDir, buildName, buildNumber strin
...
+		modTime, err := time.Parse("2006-01-02T15:04:05.999Z", item.Modified)
+		if err != nil {
+			log.Debug("Could not parse modified time for " + item.Name + ": " + err.Error())
+			continue
+		}
+
+		// Only include artifacts modified after cutoff
+		if modTime.After(cutoffTime) {
+			recentArtifacts = append(recentArtifacts, *item)
+			log.Debug(fmt.Sprintf("Including recently deployed artifact: %s (modified: %s)", item.Name, item.Modified))
```
Comment (bhanurp) [inline artifactory/commands/flexpack/maven.go]: seeing too many debug logs for build info construction. Would you like to have verbose level?

File: artifactory/commands/flexpack/maven.go [hunk 2]
```diff
@@ -144,63 +180,159 @@ func setMavenBuildPropertiesOnArtifacts(workingDir, buildName, buildNumber strin
...
 	log.Info("Successfully set build properties on deployed Maven artifacts")
 	return nil
 }
 
-// getMavenArtifactCoordinates extracts Maven coordinates from pom.xml
-func getMavenArtifactCoordinates(workingDir string) (groupId, artifactId, version string, err error) {
-	// Read pom.xml to get project information
+// getMavenDeployRepository determines where Maven deployed artifacts
+// by parsing pom.xml distributionManagement
+func getMavenDeployRepository(workingDir string) (string, error) {
```
Comment (bhanurp) [inline artifactory/commands/flexpack/maven.go]: what if the the deployment repository is mentioned in settings.xml? is it handled?
